contracts/tokenomics/generator/src/contract.rs

```diff
@@ -1728,6 +1766,17 @@ pub fn create_pool(deps: DepsMut, env: &Env, lp_token: &Addr) -> Result<PoolInfo
         }
     }
 
+    let factory_pair: PairInfo = deps.querier.query_wasm_smart(
+        cfg.factory.clone(),
+        &FactoryQueryMsg::Pair {
+            asset_infos: pair_info.asset_infos,
+        },
+    )?;
+
+    if &factory_pair.liquidity_token != lp_token {
+        return Err(ContractError::PoolDuplicate {});
+    }
+
```

Comment:
We can simply use a flag on the loop above, to check that pool's type is registered